agent/xds/clusters.go

```diff
@@ -243,8 +262,22 @@ func (s *Server) makeGatewayServiceClusters(cfgSnap *proxycfg.ConfigSnapshot) ([
 			}
 			cluster := s.makeGatewayCluster(cfgSnap, opts)
 
-			if cfgSnap.Kind == structs.ServiceKindTerminatingGateway {
+			switch cfgSnap.Kind {
```

Comment:
trivial: do you think this blob of code should be made into a small utility function since you repeat what appears to be the same thing here twice?

Reply:
Reworked this function